ods_ci/tasks/Resources/Provisioning/GPU/provision-gpu.sh

```diff
@@ -1,20 +1,72 @@
 #!/bin/bash
 set -e
 
+# GPU Node Provisioning Script for Self-Managed Clusters
+# 
+# This script creates GPU nodes in self-managed OpenShift clusters by creating
+# a new MachineSet with the specified configuration.
+#
+# Parameters:
+#   $1 INSTANCE_TYPE   - Instance type for the GPU nodes (default: g4dn.xlarge)
+#   $2 PROVIDER        - Cloud provider (AWS/GCP/AZURE/IBM) (default: AWS)
+#   $3 GPU_COUNT       - Number of GPUs per node (for GCP only) (default: 1)
+#   $4 GPU_NODE_COUNT  - Number of GPU nodes to create (default: 1)
+#
+# Examples:
+#   ./provision-gpu.sh                           # Create 1 g4dn.xlarge node on AWS
+#   ./provision-gpu.sh g4dn.2xlarge AWS 1 3     # Create 3 g4dn.2xlarge nodes on AWS
+#   ./provision-gpu.sh n1-standard-4 GCP 2 2    # Create 2 nodes with 2 GPUs each on GCP
+
 # Optional params
 INSTANCE_TYPE=${1:-"g4dn.xlarge"}
 PROVIDER=${2:-"AWS"}
 GPU_COUNT=${3:-"1"}
+GPU_NODE_COUNT=${4:-"1"}
 KUSTOMIZE_PATH="$PWD/tasks/Resources/Provisioning/Hive/GPU"
 MACHINESET_PATH="$KUSTOMIZE_PATH/base/source-machineset.yaml"
 PROVIDER_OVERLAY_DIR=$KUSTOMIZE_PATH/overlays/$PROVIDER
 MACHINE_WAIT_TIMEOUT=10m
 # Check if existing machineset GPU already exists
 EXISTING_GPU_MACHINESET="$(oc get machinesets.machine.openshift.io -n openshift-machine-api -o jsonpath="{.items[?(@.metadata.annotations['machine\.openshift\.io/GPU']>'0')].metadata.name}")"
 if [[ -n "$EXISTING_GPU_MACHINESET" ]] ; then
-  echo "Machine-set for GPU already exists"
-  oc get machinesets -A --show-labels
-  exit 0
+  echo "Machine-set for GPU already exists: $EXISTING_GPU_MACHINESET"
+  
+  # Get current replica count with proper error handling
+  CURRENT_REPLICAS=$(oc get machineset.machine.openshift.io $EXISTING_GPU_MACHINESET -n openshift-machine-api -o jsonpath='{.spec.replicas}' 2>/dev/null)
+  
+  # Check if we could get the replica count
+  if [[ -z "$CURRENT_REPLICAS" ]]; then
+    echo "Error: Could not get replica count for MachineSet $EXISTING_GPU_MACHINESET"
+    echo "MachineSet might not exist or use different API version. Listing all GPU MachineSets:"
+    oc get machinesets -n openshift-machine-api | grep -i gpu || echo "No GPU MachineSets found"
+    echo "Proceeding to create new MachineSet..."
```

Comment:
when we arrive at this line of code, it means we have already checked the existence of a GPU machine set https://github.com/red-hat-data-services/ods-ci/pull/2641/files#diff-df68ef433c7338721604845b3f7fee1550f1e2229bf267cacc00955310f95684R31